### DISCUSSION TITLE
Unique Faucet Scheme for Token Minting

### DISCUSSION THREAD

**Discussion by ElliotFriend** [stellar/sorobanathon#93] [General]
As a practice/experimental effort, I've designed a contract that allows people to "sign up" to receive an amount of a token following the Fibonacci Sequence. Each account that signs up is given more and more of the token until eventually, the whole system grinds to a halt! (perhaps a tad dramatic...)

It was interesting to code, and has *really* helped me to learn and expand my Rust/Soroban skills. Anyway, here's the repo: https://github.com/ElliotFriend/soroban-fib-faucet

Hope it doesn't suck lol 👍🏻

**Comment by achallagundla** [comment]
Heya @ElliotFriend  ,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1200 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.

**Reply by ElliotFriend** [reply]
Done. Thanks!! 🙏🏻
